### PR TITLE
Mark classes with @class to try and find implicitThis errors

### PR DESCRIPTION
**Pull request by AlCalzone**

### CODE REVIEW

File: lib/setup/setupBackup.js [hunk 1]
```diff
@@ -59,7 +59,7 @@ class BackupRestore {
                 for (let t = 0; t < res.length; t++) {
                     if (res[t].isDir) {
                         count++;
-                        copyDir(id, srcPath + '/' + res[t].file, destPath + '/' + res[t].file, () => {
+                        this.copyDir(id, srcPath + '/' + res[t].file, destPath + '/' + res[t].file, () => {
```
Comment (AlCalzone) [inline lib/setup/setupBackup.js]: caught one

File: lib/setup/setupBackup.js [hunk 2]
```diff
@@ -148,7 +148,7 @@ class BackupRestore {
         // copy
         if (fs.lstatSync(source).isDirectory()) {
             files = fs.readdirSync(source);
-            files.forEach(function (file) {
+            files.forEach((file) => {
                 const curSource = pathLib.join(source, file);
                 if (fs.lstatSync(curSource).isDirectory()) {
                     this.copyFolderRecursiveSync(curSource, targetFolder);
```
Comment (AlCalzone) [inline lib/setup/setupBackup.js]: and another one

File: lib/setup/setupInstall.js [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
+'use strict';
+
+/** @class */
 function Install(options) {
-    'use strict';
 
     const EXIT_CODES       = require('../exitCodes');
     const fs               = require('fs');
     const tools            = require('../tools.js');
-    const extend           = require('node.extend');
```
Comment (AlCalzone) [inline lib/setup/setupInstall.js]: unused

File: lib/setup/setupUpload.js [hunk 4]
```diff
@@ -13,7 +14,7 @@ function Upload(options) {
     const objects     = options.objects;
     let mime;
     // attName = file.split('/' + tools.appName + '.');
-    const regApp = new RegExp('\/' + tools.appName + '\.', 'i');
+    const regApp = new RegExp('/' + tools.appName.replace(/\./g, '\\.') + '\\.', 'i');
```
Comment (AlCalzone) [inline lib/setup/setupUpload.js]: No need to escape / in the string. To get `\.` in the RegExp, we need `\\.` in the string.

**Comment by GermanBluefox** [comment]
Nice findings

**Pull request merged**